src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -460,8 +462,12 @@ private StateEditor doTraverse(
             s1.incrementWalkDistance(getEffectiveBikeDistance());
         }
 
+        if (costExtension != null) {
+           weight += costExtension.calculateExtraCost(options, traverseMode) * length_mm/1000;
```

Comment:
```suggestion
           weight += costExtension.calculateExtraCost(options, length_mm, traverseMode);
```
I think you should pass in the `length_mm` to the extension and then do the calculation inside. Other extentions might do something completely different e.g. ignoring the length.

Reply:
Good point, I moved the calculation inside the extension.